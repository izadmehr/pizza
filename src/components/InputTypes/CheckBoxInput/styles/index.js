import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  align-items: center;
`;
export const TextInput = styled.input`
  border: none;
  font-size: 1rem;
  padding: 0 5px;
  background-color: transparent;
  caret-color: ${props => props.theme.color.primary};

  ::-ms-clear {
    display: none;
  }

  ::-webkit-input-placeholder {
    font-size: 1rem;
    font-weight: 500;
    color: #7f7f7f;
    font-family: Barlow, sans-serif;
  }

  ::placeholder {
    font-size: 1rem;
    font-weight: 500;
    color: #7f7f7f;
    font-family: Barlow, sans-serif;
  }
  :focus {
    outline: none;
  }
`;
