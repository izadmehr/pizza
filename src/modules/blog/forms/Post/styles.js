import styled from "styled-components";

import { H3 } from "../../../../components/Typography";

export const PizzaSizeSelector = styled.div`
  padding: 0 1rem;
  display: flex;
  position: relative;
  align-items: center;
  height: 2.5rem;
  border: solid 1px ${props => props.theme.color.lightGrey};
  :focus-within {
    border: solid 1px ${props => props.theme.color.primary};
  }
  width: 400px;
`;

export const InputContainer = styled.div`
  flex: 1;
`;
export const ResultsContainer = styled.ul`
  position: absolute;
  margin-top: 0.4rem;
  box-shadow: -1px 0 10px 0 rgba(0, 0, 0, 0.1), 0 0 10px 0 rgba(0, 0, 0, 0.1);
  padding: 0;
  background: white;
  list-style-type: none;
`;

export const ResultItem = styled.li`
  padding: 1rem 2rem;
  background: ${({ highlighted }) =>
    highlighted ? "rgba(0,0,0,.03)" : "white"};
  font-weight: ${({ selected }) => (selected ? "bold" : "normal")};
`;

export const PizzaSizesContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const Toppings = styled(H3)`
  margin-top: 1rem;
`;

export const TappingContainer = styled.div`
  margin-top: 0.5rem;
`;
