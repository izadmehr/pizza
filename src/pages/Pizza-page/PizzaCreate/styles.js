import styled from "styled-components";

export const PizzaSizeSelector = styled.div`
  padding: 0 1rem;
  display: flex;
  position: relative;
  align-items: center;
  height: 2.5rem;
  border: solid 1px ${props => props.theme.color.lightGrey};
  :focus-within {
    border: solid 1px ${props => props.theme.color.primary};
  }
  width: 400px;
`;

export const InputContainer = styled.div`
  flex: 1;
`;
export const ResultsContainer = styled.ul`
  position: absolute;
  margin-top: 0.4rem;
  box-shadow: -1px 0 10px 0 rgba(0, 0, 0, 0.1), 0 0 10px 0 rgba(0, 0, 0, 0.1);
  padding: 0;
  background: white;
  list-style-type: none;
`;

export const ResultItem = styled.li`
  padding: 1rem 2rem;
  background: ${({ highlighted }) =>
    highlighted ? "rgba(0,0,0,.03)" : "white"};
  font-weight: ${({ selected }) => (selected ? "bold" : "normal")};
`;
