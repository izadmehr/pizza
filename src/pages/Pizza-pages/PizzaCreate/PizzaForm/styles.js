import styled from "styled-components";

import { H3 } from "components/Typography";

export const PizzaSizesContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const Toppings = styled(H3)`
  margin-top: 1rem;
`;

export const TappingContainer = styled.div`
  margin-top: 0.5rem;
`;
