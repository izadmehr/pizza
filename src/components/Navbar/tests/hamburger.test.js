import React from "react";
import { mount, shallow } from "enzyme";

import { theme } from "styles";

import "jsdom-global/register";
import "jest-styled-components";
import HamburgerComponent, { Hamburger, HamburgerLine } from "../Hamburger";

describe("Hamburger Components", () => {
  describe("Hamburger", () => {
    it("has correct default styles", () => {
      const wrapper = mount(<Hamburger theme={theme} />);
      expect(wrapper).toMatchSnapshot();
    });

    it("has correct default styles when open", () => {
      const wrapper = mount(<Hamburger isOpen theme={theme} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("HamburgerLine", () => {
    it("has correct default styles", () => {
      const wrapper = mount(<HamburgerLine theme={theme} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("HamburgerComponent", () => {
    it("renders with all props", () => {
      const wrapper = shallow(
        <HamburgerComponent theme={theme} isOpen toggleIsOpen={() => ({})} />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
