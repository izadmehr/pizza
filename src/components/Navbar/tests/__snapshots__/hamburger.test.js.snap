// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger Components Hamburger has correct default styles 1`] = `
.c0 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin: 14px;
  width: 2rem;
  height: 2rem;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.25s ease;
  transition: 0.25s ease;
  cursor: pointer;
  display: block;
}

.c0 span:nth-child(1) {
  top: 0;
}

.c0 span:nth-child(2),
.c0 span:nth-child(3) {
  top: 0.6rem;
}

.c0 span:nth-child(4) {
  top: 1.2rem;
}

@media (min-width:62.5rem) {
  .c0 {
    display: none;
  }
}

<styled.div
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
        "minWidth": "25rem",
      },
    }
  }
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  align-self: flex-end;
  margin: 14px;
  width: 2rem;
  height: 2rem;
  transform: rotate(0deg);
  transition: 0.25s ease;
  cursor: pointer;
  display: block;

  span:nth-child(1) {
    top: 0;
  }

  span:nth-child(2),
  span:nth-child(3) {
    top: 0.6rem;
  }

  span:nth-child(4) {
    top: 1.2rem;
  }

  ",
            [Function],
            ";

  ",
            "@media (min-width: 62.5rem)",
            " {
    display: none;
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "color": Object {
          "black": "#000",
          "darkGreen": "#37bc9b",
          "darkGrey": "#aab2bd",
          "darkRed": "#da4453",
          "lightGreen": "#48cfad",
          "lightGrey": "#ccd1d9",
          "lightRed": "#ed5565",
          "primary": "#5d9cec",
          "secondary": "#4a89dc",
          "white": "#fff",
        },
        "layout": Object {
          "maxWidth": "62.5rem",
          "minWidth": "25rem",
        },
      }
    }
  >
    <div
      className="c0"
    />
  </StyledComponent>
</styled.div>
`;

exports[`Hamburger Components Hamburger has correct default styles when open 1`] = `
.c0 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin: 14px;
  width: 2rem;
  height: 2rem;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.25s ease;
  transition: 0.25s ease;
  cursor: pointer;
  display: block;
}

.c0 span:nth-child(1) {
  top: 0;
}

.c0 span:nth-child(2),
.c0 span:nth-child(3) {
  top: 0.6rem;
}

.c0 span:nth-child(4) {
  top: 1.2rem;
}

.c0 span:nth-child(1) {
  top: 0.8rem;
  width: 0%;
  left: 50%;
}

.c0 span:nth-child(2) {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c0 span:nth-child(3) {
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.c0 span:nth-child(4) {
  top: 0.8rem;
  width: 0%;
  left: 50%;
}

@media (min-width:62.5rem) {
  .c0 {
    display: none;
  }
}

<styled.div
  isOpen={true}
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
        "minWidth": "25rem",
      },
    }
  }
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  align-self: flex-end;
  margin: 14px;
  width: 2rem;
  height: 2rem;
  transform: rotate(0deg);
  transition: 0.25s ease;
  cursor: pointer;
  display: block;

  span:nth-child(1) {
    top: 0;
  }

  span:nth-child(2),
  span:nth-child(3) {
    top: 0.6rem;
  }

  span:nth-child(4) {
    top: 1.2rem;
  }

  ",
            [Function],
            ";

  ",
            "@media (min-width: 62.5rem)",
            " {
    display: none;
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    isOpen={true}
    theme={
      Object {
        "color": Object {
          "black": "#000",
          "darkGreen": "#37bc9b",
          "darkGrey": "#aab2bd",
          "darkRed": "#da4453",
          "lightGreen": "#48cfad",
          "lightGrey": "#ccd1d9",
          "lightRed": "#ed5565",
          "primary": "#5d9cec",
          "secondary": "#4a89dc",
          "white": "#fff",
        },
        "layout": Object {
          "maxWidth": "62.5rem",
          "minWidth": "25rem",
        },
      }
    }
  >
    <div
      className="c0"
    />
  </StyledComponent>
</styled.div>
`;

exports[`Hamburger Components HamburgerComponent renders with all props 1`] = `
<styled.div
  isOpen={true}
  onClick={[Function]}
>
  <styled.span
    isOpen={true}
  />
  <styled.span
    isOpen={true}
  />
  <styled.span
    isOpen={true}
  />
  <styled.span
    isOpen={true}
  />
</styled.div>
`;

exports[`Hamburger Components HamburgerLine has correct default styles 1`] = `
.c0 {
  display: block;
  position: absolute;
  border-radius: 1.6rem;
  height: 0.2rem;
  width: 100%;
  opacity: 1;
  left: 0;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.25s ease-in-out;
  transition: 0.25s ease-in-out;
  background: #fff;
}

@media (min-width:48rem) {
  .c0 {
    background: #fff;
  }
}

<styled.span
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
        "minWidth": "25rem",
      },
    }
  }
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  display: block;
  position: absolute;
  border-radius: 1.6rem;
  height: 0.2rem;
  width: 100%;
  opacity: 1;
  left: 0;
  transform: rotate(0deg);
  transition: 0.25s ease-in-out;
  background: ",
            [Function],
            ";

  ",
            "@media (min-width: 48rem)",
            " {
    background: ",
            [Function],
            ";
  }
",
          ],
        },
        "displayName": "styled.span",
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "span",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "color": Object {
          "black": "#000",
          "darkGreen": "#37bc9b",
          "darkGrey": "#aab2bd",
          "darkRed": "#da4453",
          "lightGreen": "#48cfad",
          "lightGrey": "#ccd1d9",
          "lightRed": "#ed5565",
          "primary": "#5d9cec",
          "secondary": "#4a89dc",
          "white": "#fff",
        },
        "layout": Object {
          "maxWidth": "62.5rem",
          "minWidth": "25rem",
        },
      }
    }
  >
    <span
      className="c0"
    />
  </StyledComponent>
</styled.span>
`;
